cscs-checks/microbenchmarks/cpu/likwid/likwid.py

```diff
@@ -8,85 +8,81 @@
+    num_tasks_per_core = 2
+    system_num_cpus = {
+        'daint:mc':  72,
+        'daint:gpu': 24,
+        'dom:mc':  72,
+        'dom:gpu': 24,
+    }
+    system_numa_domains = {
+        'daint:mc':  ['S0', 'S1'],
+        'daint:gpu': ['S0'],
+        'dom:mc':  ['S0', 'S1'],
+        'dom:gpu': ['S0'],
+    }
+    # Test each level at half capacity times nthreads per domain
+    system_cache_sizes = {
+        'daint:mc':  {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                      'memory': '1800MB'},
+        'daint:gpu': {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                      'memory': '1200MB'},
+        'dom:mc':    {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                      'memory': '1800MB'},
+        'dom:gpu':   {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                      'memory': '1200MB'},
+    }
+    maintainers = ['SK', 'CB']
+    tags = {'benchmark', 'diagnostic', 'health'}
+
+    @sanity_function
+    def assert_ge(self):
+        self.bw_pattern = sn.min(sn.extractall(r'MByte/s:\s*(?P<bw>\S+)',
+                                               self.stdout, 'bw', float))
+        return sn.assert_ge(self.bw_pattern, 0.0)
+
+    @performance_function('MB/s')
+    def bandwidth(self):
+        return self.bw_pattern
 
-        self.executable = 'likwid-bench'
 
-        self.num_tasks = 1
-        self.num_tasks_per_node = 1
-        self.num_tasks_per_core = 2
-        self.system_num_cpus = {
-            'daint:mc':  72,
-            'daint:gpu': 24,
-            'dom:mc':  72,
-            'dom:gpu': 24,
-        }
-        self.system_numa_domains = {
-            'daint:mc':  ['S0', 'S1'],
-            'daint:gpu': ['S0'],
-            'dom:mc':  ['S0', 'S1'],
-            'dom:gpu': ['S0'],
-        }
-
-        # Test each level at half capacity times nthreads per domain
-        self.system_cache_sizes = {
-            'daint:mc':  {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
-                          'memory': '1800MB'},
-            'daint:gpu': {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
-                          'memory': '1200MB'},
-            'dom:mc':    {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
-                          'memory': '1800MB'},
-            'dom:gpu':   {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
-                          'memory': '1200MB'},
-        }
-
-        self.maintainers = ['SK', 'CB']
-        self.tags = {'benchmark', 'diagnostic', 'health'}
-
-        bw_pattern = sn.min(sn.extractall(r'MByte/s:\s*(?P<bw>\S+)',
-                                          self.stdout, 'bw', float))
-
-        self.sanity_patterns = sn.assert_ge(bw_pattern, 0.0)
-        self.perf_patterns = {
-            'bandwidth': bw_pattern
-        }
-
-
-@rfm.parameterized_test(*[[l, k] for l in ['L1', 'L2', 'L3']
+@rfm.simple_test
+class CPUBandwidth(MemBandwidthTest):
+    config = parameter([*[[l, k] for l in ['L1', 'L2', 'L3']
```

Comment:
I prefer these to be different parameters and have reframe do the external product.

Reply:
I think that can't be done here since the combinations are only of `['L1', 'L2', 'L3']` and  `['load_avx', 'store_avx']`. The other two variants `['memory', 'load_avx']` and `['memory', 'store_mem_avx']` are added at the end.